### PR TITLE
Feat/check reservations

### PR DESCRIPTION
**Pull request by albertmakan**

### CODE REVIEW

File: ftrip.io.catalog-service/ftrip.io.catalog-service/Installers/DependenciesInstaller.cs [hunk 1]
```diff
@@ -22,6 +22,7 @@ public void Install()
             _services.AddProxiedScoped<IAmenityRepository, AmenityRepository>();
             _services.AddProxiedScoped<IAmenityTypeRepository, AmenityTypeRepository>();
             _services.AddProxiedScoped<IPropertyTypeRepository, PropertyTypeRepository>();
+            _services.AddProxiedScoped<IBookingServiceClient, BookingServiceClient>();
```
Comment (panicmilos) [inline ftrip.io.catalog-service/ftrip.io.catalog-service/Installers/DependenciesInstaller.cs]: Http services should be added with "AddScoped". There is already built-in tracer that will open new tracing session.

File: ftrip.io.catalog-service/ftrip.io.catalog-service/Startup.cs [hunk 2]
```diff
@@ -38,6 +38,9 @@ public Startup(IConfiguration configuration)
         public void ConfigureServices(IServiceCollection services)
         {
             services.AddControllers();
+            services.AddHttpClient("booking", client =>
+                client.BaseAddress = new Uri(Environment.GetEnvironmentVariable("BOOKING_SERVICE_URL"))
```
Comment (panicmilos) [inline ftrip.io.catalog-service/ftrip.io.catalog-service/Startup.cs:42]: Nice, you are using best practices. We should figure out how to add Correlation Id and Jwt Token when sending HTTP requests.

File: ftrip.io.catalog-service/ftrip.io.catalog-service/Accommodations/UseCases/CreateAccommodation/CreateAccommodationRequestHandler.cs [hunk 3]
```diff
@@ -23,6 +24,7 @@ public class CreateAccommodationRequestHandler : IRequestHandler<CreateAccommoda
         private readonly IMapper _mapper;
         private readonly IStringManager _stringManager;
         private readonly IMessagePublisher _messagePublisher;
+        private readonly ILogger _logger;
 
         public CreateAccommodationRequestHandler(
```
Comment (panicmilos) [inline ftrip.io.catalog-service/ftrip.io.catalog-service/Accommodations/UseCases/CreateAccommodation/CreateAccommodationRequestHandler.cs:29]: I guess we can increase num of parameters. Here, we are injecting 4 utility classes so I feel like this handler is still fine.

**Review by panicmilos — CHANGES_REQUESTED** [review]
Great job. One small change is needed.

File: ftrip.io.catalog-service/ftrip.io.catalog-service/Startup.cs (hunk 2 above)
Comment (albertmakan) [reply ftrip.io.catalog-service/ftrip.io.catalog-service/Startup.cs:42]: We can add Jwt like this:
client.DefaultRequestHeaders.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue("...");

Comment (panicmilos) [reply ftrip.io.catalog-service/ftrip.io.catalog-service/Startup.cs:42]: That is true. Only question is how to get instances of contexts but we have services there so we can also get service provider for sure.

**Review by panicmilos — APPROVED** [review]

**Pull request merged**